### PR TITLE
dtj chapter 16

### PR DESCRIPTION
**Pull request by thatguy59**

### CODE REVIEW

File: chapter16/dtj/ex2 [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+# Probably not much to gain by updating the previous one here...
+
+use strict;
+use warnings;
+
+open my $std_out, '>', 'ls.out' or die "Could not open output! $!";
+open my $std_err, '>', 'ls.err' or die "Could not open output! $!";
+
+*STDERR = $std_err;
```
Comment (MarkWignall) [inline chapter16/dtj/ex2:11]: What does the \* do?

File: chapter16/dtj/ex2 [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+do_command("cd / && ls -l");
+do_command("ThisIsNotARealCommand");
```
Comment (MarkWignall) [inline chapter16/dtj/ex2:14]: Does "ThisIsNotARealCommand" get printed?

File: chapter16/dtj/ls.err [hunk 3]
```diff
@@ -0,0 +1 @@
+Can't exec "ThisIsNotARealCommand": No such file or directory at ./ex2 line 18.
```
Comment (MarkWignall) [inline chapter16/dtj/ls.err:1]: Well that answers my above question. =)

File: chapter16/dtj/ex2 (hunk 2 above)
Comment (LordRafa) [inline chapter16/dtj/ex2]: I think that it is to generate an error

File: chapter16/dtj/ex2 [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+use strict;
+use warnings;
+
+open my $std_out, '>', 'ls.out' or die "Could not open output! $!";
+open my $std_err, '>', 'ls.err' or die "Could not open output! $!";
+
+*STDERR = $std_err;
+
+do_command("cd / && ls -l");
```
Comment (LordRafa) [inline chapter16/dtj/ex2:13]: The functions system or exec would had been an alternative to avoid the print

File: chapter16/dtj/ex4 [hunk 5]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use Data::Dumper;
+
+my %sigs;
```
Comment (LordRafa) [inline chapter16/dtj/ex4:7]: You could declare this inside of the function as state (static in perl)

File: chapter16/dtj/ex2 (hunk 2 above)
Comment (qianliok) [inline chapter16/dtj/ex2:14]: Good testing coverage! :+1:

File: chapter16/dtj/ex2 (hunk 1 above)
Comment (alexdeas) [inline chapter16/dtj/ex2:11]: That's how you update the symbol table, probably best not to mess around with it at the moment.

**Pull request merged**